jetbrains.*: update versions

sdht0 opened the pull request:
## Description of changes

* Updated using `bin/update_bin.py`.
* Fixed rust-rover build, closes #292196
* Tested rust-rover, idea, goland
* Would also be nice to merge #285710 

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

sdht0 committed:
jetbrains.rust-rover: fix build by removing unneeeded patchelf

SebTM reviewed (approved):
14 packages built:
jetbrains.clion jetbrains.dataspell jetbrains.gateway jetbrains.goland jetbrains.idea-community jetbrains.idea-community-bin jetbrains.idea-ultimate jetbrains.phpstorm jetbrains.pycharm-community jetbrains.pycharm-community-bin jetbrains.pycharm-professional jetbrains.ruby-mine jetbrains.rust-rover jetbrains.webstorm

Result of `nixpkgs-review pr 293308` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>jetbrains.clion</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.gateway</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-src)</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community (jetbrains.pycharm-community-src)</li>
    <li>jetbrains.pycharm-community-bin</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>

Tested phpstorm locally works fine ✌🏻

niknetniko reviewed (approved):
Intellij Ultimate works for me.

sdht0 commented:
Meanwhile new version are out. Might as well update.

lhanson commented:
[like]  Lyle Hanson reacted to your message:
________________________________
From: Siddhartha Sahu ***@***.***>
Sent: Friday, March 8, 2024 3:11:24 PM
To: NixOS/nixpkgs ***@***.***>
Cc: Lyle Hanson ***@***.***>; Manual ***@***.***>
Subject: Re: [NixOS/nixpkgs] jetbrains.*: update versions (PR #293308)


Meanwhile new version are out. Might as well update.

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/NixOS/nixpkgs/pull/293308*issuecomment-1985866194__;Iw!!Mak6IKo!JXYaAA8-B8uro9fHQ2kk-ETKyvZjWs9OsgqClt0cI8H8fihedODvfFHaCEZUQq-8grE9SRif9tLA12vQ23N7OQxmdos$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AACCY7NAI53CH5YUEW2UXO3YXHIJZAVCNFSM6AAAAABEFXG2J6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBVHA3DMMJZGQ__;!!Mak6IKo!JXYaAA8-B8uro9fHQ2kk-ETKyvZjWs9OsgqClt0cI8H8fihedODvfFHaCEZUQq-8grE9SRif9tLA12vQ23N7XWfNNS4$>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

sdht0 committed:
jetbrains: 2023.2 -> 2023.3.5

jetbrains.clion: 2023.3.3 -> 2023.3.4
jetbrains.dataspell: 2023.3.3 -> 2023.3.4
jetbrains.gateway: 2023.3.3 -> 2023.3.4
jetbrains.goland: 2023.3.3 -> 2023.3.5
jetbrains.idea-community: 2023.3.3 -> 2023.3.5
jetbrains.idea-ultimate: 2023.3.3 -> 2023.3.5
jetbrains.mps: 2023.2 -> 2023.3
jetbrains.phpstorm: 2023.3.3 -> 2023.3.5
jetbrains.pycharm-community: 2023.3.3 -> 2023.3.4
jetbrains.pycharm-professional: 2023.3.3 -> 2023.3.4
jetbrains.rider: 2023.3.3 -> 2023.3.4
jetbrains.ruby-mine: 2023.3.3 -> 2023.3.5
jetbrains.rust-rover: 2023.3 EAP -> 2023.3 EAP
jetbrains.webstorm: 2023.3.3 -> 2023.3.4

sdht0 committed:
jetbrains.plugins: update

sdht0 commented:
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>jetbrains.clion</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.gateway</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-src)</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.mps</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community (jetbrains.pycharm-community-src)</li>
    <li>jetbrains.pycharm-community-bin</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3623

liff commented on pkgs/applications/editors/jetbrains/default.nix:
```diff
@@ -240,8 +240,6 @@ rec {
 
-        interp="$(cat $NIX_CC/nix-support/dynamic-linker)"
-        patchelf --set-interpreter $interp $PWD/plugins/intellij-rust/bin/linux/*/intellij-rust-native-helper
         chmod +x $PWD/plugins/intellij-rust/bin/linux/*/intellij-rust-native-helper
```
Not that there’s any harm in having this, but it looks like in the latest release both the x86-64 and arm64 binaries have the executable bit set.

liff reviewed (approved):
Rust Rover works for me.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1542
